sonar-batch/src/main/java/org/sonar/batch/report/MeasuresPublisher.java

```diff
@@ -86,13 +87,18 @@ private void setValueAccordingToType(BatchReport.Measure.Builder builder, Measur
         builder.setBooleanValue((Boolean) value);
         break;
       case DOUBLE:
-        builder.setDoubleValue((Double) value);
+        builder.setDoubleValue(((Number) value).doubleValue());
         break;
       case INT:
-        builder.setIntValue((Integer) value);
+        builder.setIntValue(((Number) value).intValue());
         break;
       case LONG:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)